### DISCUSSION TITLE
Is it possible for me to edit any of the mod's files to change effect duration times?

### DISCUSSION THREAD

**Discussion by sumsare** [gta-chaos-mod/Trilogy-ASI-Script#321] [Q&A]
For example, if I wanted to extend the buttsbot effect to like 5-10 minutes, would I need a program to do that? Or just edit something in Notepad++? Or would I have to download the source files and edit them in an IDE?

**Comment by Lordmau5** [comment]
In the current system you would need to update the GUI, yes.
https://github.com/gta-chaos-mod/Trilogy-GUI-Tool

Specifically this one
https://github.com/gta-chaos-mod/Trilogy-GUI-Tool/blob/v3.0/GTAChaos/src/effects/EffectDatabase.cs#L285

The last 2 parameters (-1 and 1.0f) are the duration and the modifier respectively.
If you'd want to do a set duration you'd set -1 to, say, 60, which would mean Buttsbot would *always* be 60 seconds long.
If you keep it at -1 and only change the modifier from 1.0f to, say, 3.0f that'd mean it'd be 3x as long (so cooldown * 3)

I did want to allow overriding durations for effects in some way or form but couldn't think of a good way to do that.

**Reply by sumsare** [reply]
Thank you for the reply! However, I have one follow-up question. I can't figure out what to do with the GUI Tool once I've downloaded it (since there was no release, I just downloaded the entire source as a zip and then unpacked it) and the Readme is supposed to be part of this repo here, but when I do a Ctrl + F for "GUI", it's only referenced in two places and doesn't say how to use the tool. Could you please tell me what to do in order to use the Tool in the way you described? Do I have to unpack it into my GTA SA game folder or launch separately somehow?

**Comment by Lordmau5** [comment]
The GUI tool is a Visual Studio project, so you'll need to download and install that, then build the solution (which is the internal name for the project) and it'll give you an exe.

https://visualstudio.microsoft.com/vs/community/

**Reply by sumsare** [reply]
I see, thank you very much again :)
